mobile/test/common/integration/client_integration_test.cc

```diff
@@ -1360,5 +1361,15 @@ TEST_P(ClientIntegrationTest, TestProxyResolutionApi) {
 }
 #endif
 
+TEST_P(ClientIntegrationTest, OnNetworkChanged) {
+  builder_.addRuntimeGuard("dns_cache_set_ip_version_to_remove", true);
+  initialize();
+  internalEngine()->setPreferredNetwork(NetworkType::WLAN);
+  basicTest();
```

Comment:
i'm not 100% clear on what this test is trying to test when we change the preferred network to WLAN?

Reply:
This is more to exercise the IP version removal code + DNS refresh so that it doesn't crash. Unfortunately, it's not easy to really simulate switching from one network type to another network type. Comment updated to verify what this test is for.